### DISCUSSION TITLE
Leap Motion API Support

### DISCUSSION THREAD

**Discussion by plastegiraffe** [alvr-org/ALVR#2518] [General]
Recently, hand tracking was officially supported in VRChat. Since I have a Pico 4 and the hand tracking works rather below average there, I bought a Leap Motion 2 controller. I was actually planning to stick the Leap Motion 2 to my Pico 4 with a 3D printed mount and then stream it to the PC via VirtualHere. Unfortunately, this doesn't work because the Leap Motion software requires the device to be connected directly to the PC. To test this, I sat down with a 5m USB cable and tried it anyway. Even though the software reports that my device is only connected with USB 2.0 and therefore the tracking quality is reduced, the result was quite good.

But now Leap Motion also provides APK files for the Quest and the Pico to connect them directly. I have tested it in standalone mode and it also works there without any problems. However, apart from a few demo programs, there is currently nothing that can interact with the API there.

Is it possible to integrate this API into ALVR so that I can use the Leap Motion in VRChat?
This way, I wouldn't have to run a long cable through the living room and could connect the Leap Motion directly to the VR.

The APK's for standalone operation are provided here:

https://leap2.ultraleap.com/downloads/leap-motion-controller-2/

and the API is documented here:

https://blog.leapmotion.com/getting-started-leap-motion-sdk/

**Comment by zmerp** [comment]
I think that maybe @ShootingKing-AM could be interested in working on this. For the main ALVR project this would be rather low priority

**Comment by ShootingKing-AM** [comment]
I have a LeapController myself, but, this is a low priority even for PhoneVR. Since not everyone would have a Leapmotion device.
